### PR TITLE
Update .git attr.

### PR DESCRIPTION
**Pull request by SpacePossum**
replaces
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2145

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -5,3 +5,6 @@
 .travis.yml export-ignore
 appveyor.yml export-ignore
 phpunit.xml.dist export-ignore
+*.php   text eol=lf whitespace=blank-at-eol,blank-at-eof,space-before-tab,tab-in-indent,tabwidth=4 diff=php
+*.json  text eol=lf whitespace=blank-at-eol,blank-at-eof,space-before-tab,tab-in-indent,tabwidth=4
+*.yml   text eol=lf whitespace=blank-at-eol,blank-at-eof,space-before-tab,tab-in-indent,tabwidth=4
```
Comment (keradus) [inline .gitattributes]: please add blank lind

**Comment by keradus** [comment]
I would do `eol=lf` for all text files

**Comment by SpacePossum** [comment]
done.... but thinking about it, this will be applied to the '.test' files as well now right? We might want to allow other formatting there because of test purposes.

File: .gitattributes [hunk 2]
```diff
@@ -5,6 +5,7 @@
 .travis.yml export-ignore
 appveyor.yml export-ignore
 phpunit.xml.dist export-ignore
+*       text=auto eol=lf
 *.php   text eol=lf whitespace=blank-at-eol,blank-at-eof,space-before-tab,tab-in-indent,tabwidth=4 diff=php
```
Comment (keradus) [inline .gitattributes]: no need to respecify eol again I guess?

Comment (SpacePossum) [inline .gitattributes]: not sure, let me try some stuff out

Comment (keradus) [inline .gitattributes]: did it work ?

**Comment by SpacePossum** [comment]
This messes up tests that have diff. code style by design, for example tabs in heredocs. It also effects `.test` integration tests. All-in-all it is not worth it IMHO.

**Comment by keradus** [comment]
@SpacePossum 
what do you mean by `messes up` ? it's autocorrects them or requires you to add to staging with force ?

**Comment by SpacePossum** [comment]
```bash
possum@milkyway:~/work/PHP-CS-Fixer$ php test.php
possum@milkyway:~/work/PHP-CS-Fixer$ git add test.php 
possum@milkyway:~/work/PHP-CS-Fixer$ git add test2.php 
warning: CRLF will be replaced by LF in test2.php.
The file will have its original line endings in your working directory.
possum@milkyway:~/work/PHP-CS-Fixer$ git st
On branch 1_12_update_git_attr
Your branch is up-to-date with 'origin/1_12_update_git_attr'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	new file:   test.php
	new file:   test2.php

possum@milkyway:~/work/PHP-CS-Fixer$ git ci 'test'
[1_12_update_git_attr d264129] test
warning: CRLF will be replaced by LF in test2.php.
The file will have its original line endings in your working directory.
 2 files changed, 14 insertions(+)
 create mode 100644 test.php
 create mode 100644 test2.php
possum@milkyway:~/work/PHP-CS-Fixer$ git push
```

looking at the diff on github I think not was applied as expected

**Comment by keradus** [comment]
with `core.safecrlf = true` :

```
D:\GIT\fork\PHP-CS-Fixer (tmp)
λ git add test2.php
fatal: CRLF would be replaced by LF in test2.php.

D:\GIT\fork\PHP-CS-Fixer (tmp)
λ git add --force test2.php
fatal: CRLF would be replaced by LF in test2.php.
```

so, sadly, pointless. usually we want to keep proper line ending and so on, but we also add bad one in fixtures, so we need a way to have both ways in one repo.

Sadly, closing.

**Comment by SpacePossum** [comment]
thanks for checking @keradus !

**Pull request closed without merging**